pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/QueryRunner.java

```diff
@@ -261,15 +264,15 @@ public void processTimeSeriesQuery(String serializedPlan, Map<String, String> me
       responseObserver.onCompleted();
     };
     try {
-      final long timeoutMs = extractTimeoutMs(metadata);
-      Preconditions.checkState(timeoutMs > 0,
-          "Query timed out before getting processed in server. Remaining time: %s", timeoutMs);
+      final long deadlineMs = extractDeadlineMs(metadata);
```

Comment:
Is it a timeout or deadline? If it is timeout then the pre check below seems odd?

Reply:
It is deadline epoch millis